Mage/Archivist/Archivist.cs

```diff
@@ -133,12 +133,18 @@ public class Archivist : EventEmitter<JObject> {
 			//if (result == null && !options ["optional"]) {
 			//	return cb (new Exception ("ValueNotFound"), null);
 			//}
-			
-			// Create vault value
-			VaultValue newValue = new VaultValue((JObject)result);
-			
-			// Add value to cache
-			_cache.Add (cacheKeyName, newValue);
+
+			VaultValue newValue;
+			try {
+				// Create vault value
+				newValue = new VaultValue((JObject)result);
+				
+				// Add value to cache
+				_cache.Add (cacheKeyName, newValue);
+			} catch (Exception cacheError) {
+				cb(cacheError, null);
```

Comment:
I could, but that's an application level callback, meaning the user would have to make sure their callback has that signature. This is a sure fire way to prevent errors if they don't and it removes some the nuisance of having to think about it when you use this function.